cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -222,6 +222,46 @@ TEST_F(TestParquetFileFormat, WriteRecordBatchReaderCustomOptions) {
                     *actual_schema);
 }
 
+#ifdef ARROW_WITH_SNAPPY
+
+TEST_F(TestParquetFileFormat, WriteRecordBatchReaderByteStreamSplit) {
```

Comment:
Is it necessary to use datasets for this test, or can the test be implemented in the regular Parquet C++ test suite?

Reply:
I'll let @westonpace comment on whether this test is relevant for the datasets layer.

Reply:
If the feature is well tested in the `file_deserialize_test.cc` then I think we should remove this.  Even though it was found using datasets I don't think the root cause was in datasets and it doesn't appear that there are any changes to `file_parquet.cc`.

Reply:
Okay, changes are removed.